### ISSUE TITLE
json serializer incorrectly displays integer64

### ISSUE THREAD

**Issue by jasonheffner** [rstudio/plumber#904]
When using the default json serializer with integer64 values greater than 16 digits the numbers will get rounded. A simple solution has been to convert the values to character. This has become somewhat cumbersome.

### System details

```
R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] memoise_2.0.1      glue_1.6.2         stringr_1.5.0      DBI_1.1.3          future_1.31.0      R.utils_2.12.2    
 [7] R.oo_1.25.0        R.methodsS3_1.8.2  bit64_4.0.5        bit_4.0.5          lubridate_1.9.2    data.table_1.14.6 
[13] plumber_1.2.1.9000 RPostgres_1.4.5   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10       compiler_4.2.2    later_1.3.0       remotes_2.4.2     tools_4.2.2       digest_0.6.31     jsonlite_1.8.4   
 [8] lifecycle_1.0.3   timechange_0.2.0  pkgconfig_2.0.3   rlang_1.0.6       cli_3.6.0         curl_5.0.0        parallel_4.2.2   
[15] fastmap_1.1.0     swagger_3.33.1    generics_0.1.3    vctrs_0.5.2       globals_0.16.2    hms_1.1.2         listenv_0.9.0    
[22] R6_2.5.1          parallelly_1.34.0 blob_1.2.3        magrittr_2.0.3    promises_1.2.0.1  codetools_0.2-18  ellipsis_0.3.2   
[29] webutils_1.1      httpuv_1.6.9      stringi_1.7.12    cachem_1.0.6     
```

### Example application *or* steps to reproduce the problem

Given a simple test API call with hard-coded values. 

```R

library(plumber)
library(data.table)
library(bit64)

# plumber API definition

#' Return test table 
#' @param userID access ID of the user making the query
#' @param courseID course number
#' @get /test
getTestData<- function(userID,courseID){

  course<-NULL
  setDT(course)
  course$name <- "Test course"
  # Test 17 digit integer64 value
  course$course_id <- '10500000002207887'
  course$course_id <- bit64::as.integer64(course$course_id)

  # send data.table
  course 
  
  # list test
  #out <- list()
  #out$table <- course
  #out
}
```

### Expected Behavior
Returns the correct value for an integer64. Expected result is:
```
[
  {
    "name": "Test course",
    "course_id": 10500000002207887
  }
]
```
instead, we get:
```
[
  {
    "name": "Test course",
    "course_id": 10500000002207888
  }
]
```
If we attempt using toJSON directly it returns the correct result as well
```
library(jsonlite)
  jsoncars <- toJSON(course, pretty=TRUE)
  cat(jsoncars)
```

**Comment by schloerke** [comment]
What's weird is I can not reproduce it interactively, but the response from the plumber API is reproducible.

Even adding a break point using 
```r
library(plumber)
debugonce(plumber:::serializer_identity_)
pr(file = "plumber.R") %>% pr_run(port = 6621)
```

Looking at `res$toResponse()` just before `plumber:::serializer_identity_` returns...

```
> res$toResponse()
#> $status
#> [1] 200
#> 
#> $headers
#> $headers$`Content-Type`
#> [1] "application/json"
#> 
#> 
#> $body
#> [{"name":"Test course","course_id":10500000002207887}] 

> str(res$body)
#>  'json' chr "[{\"name\":\"Test course\",\"course_id\":10500000002207887}]"
```

But when the same response returns in the swagger UI, it states `...7888` and not `...7887` (expected).

**Comment by schloerke** [comment]
So I believe this is a browser bug. 

Looking at the network traffic in the swagger UI, I can see the `/test` route being executed.  The `Preview` tab has `...7888`, while the `Response` tab has `...7887`, which is correct. 

Typing the response into the JS console, produces the unexpected behavior:

![Screenshot 2023-02-21 at 11 42 55 AM](https://user-images.githubusercontent.com/93231/220406842-ca1e04c6-8356-4d98-ab67-68b94ac124a3.png)

**Comment by schloerke** [comment]
I believe plumber is doing the correct thing, just the browser/JavaScript can't handle it properly.

Even if we were able to unpack the JSON in the browser to the value, the JavaScript wouldn't be able to handle it. This would require a browser side library (such as https://www.npmjs.com/package/bignumber.js). 

Since the value being returned is valid, I believe this bug is outside the scope of the `{plumber}`. Closing

--------------------

Fun bug!!
